library/src/amd_detail/hipblaslt.cpp

```diff
@@ -402,80 +402,22 @@ try
 {
     hipblasStatus_t return_status = HIPBLAS_STATUS_SUCCESS;
 
-    if(((rocblaslt_matmul_desc)matmul_descr)->amax_ptr == nullptr)
-    {
-        return_status
-            = RocBlasLtStatusToHIPStatus(rocblaslt_matmul((rocblaslt_handle)handle,
-                                                          (rocblaslt_matmul_desc)matmul_descr,
-                                                          alpha,
-                                                          A,
-                                                          (rocblaslt_matrix_layout)matA,
-                                                          B,
-                                                          (rocblaslt_matrix_layout)matB,
-                                                          beta,
-                                                          C,
-                                                          (rocblaslt_matrix_layout)matC,
-                                                          D,
-                                                          (rocblaslt_matrix_layout)matD,
-                                                          (const rocblaslt_matmul_algo*)algo,
-                                                          workspace,
-                                                          workspaceSizeInBytes,
-                                                          stream));
-    }
-    else //Amax path
-    {
-        rocblaslt_matrix_layout tmp_matC = (rocblaslt_matrix_layout)matC;
-        rocblaslt_matrix_layout tmp_matD = (rocblaslt_matrix_layout)matD;
-        size_t amax_workspace_size       = tmp_matD->m * tmp_matD->n * 4; //only support fp32 D temp
-        void*  scaleD                    = ((rocblaslt_matmul_desc)matmul_descr)->scaleD;
-        if(!(tmp_matD->type == HIP_R_8F_E4M3_FNUZ || tmp_matD->type == HIP_R_8F_E5M2_FNUZ)
-           || tmp_matD->m != tmp_matD->ld || *(float*)beta != 0 || tmp_matD->batch_count > 1
-           || workspaceSizeInBytes < amax_workspace_size || scaleD == nullptr)
-            return HIPBLAS_STATUS_INTERNAL_ERROR;
-        hipDataType c_type                            = tmp_matD->type;
-        hipDataType d_type                            = tmp_matD->type;
-        void*       D_TEMP                            = workspace;
-        tmp_matC->type                                = HIP_R_32F;
-        tmp_matD->type                                = HIP_R_32F;
```

Comment:
remove the hidden branches for amaxD = on/off. Since we can support it by one single kernel